common/app/views/fragments/containers/index.scala.html

```diff
@@ -11,7 +11,7 @@
 @* TODO Consolidate this with the master container template to get rid of repetition *@
 
 @defining(page.webTitle) { case (webTitle) =>
-    <section class="@GetClasses.forTagContainer(webTitle.nonEmpty)"
+    <section class="@GetClasses.forTagContainer(webTitle.nonEmpty, adFree = request.headers.keys.exists(_ equalsIgnoreCase "X-Gu-Commercial-Ad-Free"))"
```

Comment:
Quite probably. I'm not sure the request implicit is passed deep enough into the backend code to be universally available - there were places it didn't seem to be referenced, but at this point, yes it could definitely read a property from there.